Utils/Dataflow/009_oracleConnector/OffsetStorage.py

```diff
@@ -0,0 +1,60 @@
+"""
+Classes representing Offset Storage.
+Provides methods:
+- init (create and open the storage)
+- get (get current offset)
+- commit (write new offset)
+"""
+
+import os.path
+import sys
+
+
+class OffsetStorage(object):
+    """ Interface class for Offset Storage. """
+    def __init__(self):
+        """ Create and open storage. """
+        raise NotImplementedError
+
+    def get(self):
+        """ Get current offset. """
+        raise NotImplementedError
+
+    def commit(self, new_offset):
+        """ Save new offset. """
+        raise NotImplementedError
+
+
+class FileOffsetStorage(OffsetStorage):
+    """ Offset Storage based on local file. """
+
+    storage = None
+    current = None
+
+    def __init__(self, filename):
+        """ Create and open storage. """
+        try:
+            if os.path.exists(filename):
+                self.storage = open(filename, 'r+', 0)
+            else:
+                self.storage = open(filename, 'w+', 0)
+        except IOError, err:
+            sys.stderr("(ERROR) Failed to initialize offset storage (%s)\n"
+                       % self.__class__.__name__)
```

Comment:
'err' local variable is never used. Can it be removed?

Reply:
[This](https://github.com/PanDAWMS/dkb/pull/93/commits/52a35decb5bad754eea69160f10cbddf8b45b0ab?diff=unified#diff-4c221825c8bb3980b06a244fe9499716R457) *if* statement looks repetitive. Maybe it would be better to create *def* like `phys_short_search(taskname, category)` and bind it with a dict of elements for checking.

I suppose, with new *def* we can check every 'taskname' in a loop using two or three strings (what if the number of task names for checking will expand in the future?).

Maybe it is not really urgent now to do this improvement. Just an assumption...

Reply:
`err` was added just automaticaly, without second thought. But I think it would be better to output error, then remove varisble ;)

Replacing `if`s is also a good idea -- will do it right now, as I`m afraid I can simply lose it later...(

Thank you!

Reply:
*Update*
Removed `err` -- as the exception is re-raised and the error message is to be output in the external code frame.

Just realized, that I just can`t replace `if` part here -- as it was moved to Stage 16 already.
Will do my best not to forget about it and address in the nearest future.